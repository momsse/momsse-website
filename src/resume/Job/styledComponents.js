import styled from 'styled-components';

export const Container = styled.div`
  padding: 0.5rem;
  background-color: #F0F0F0;
  border: 1px solid #d4d4d4;
  box-shadow: 0 1px 6px rgba(0, 0, 0, 0.175);
`;

export const Title = styled.div`
  font-size: 1.25rem;
  margin-bottom: 0.25rem;
`;

export const Period = styled.div`
  color: #7f807a;
  font-size: 0.75rem;
  margin-bottom: 0.5rem;
`
