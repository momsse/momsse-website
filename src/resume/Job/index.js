import React, { PropTypes } from 'react'
import Calendar from 'react-icons/lib/fa/calendar'
import moment from 'moment';

moment.locale('fr-FR');

import { Container, Period, Title } from './styledComponents'

const DATE_FORMAT = 'MMMM YYYY';

export function formatPeriod(startDate, endDate) {
  if (!endDate) return moment(startDate).format(DATE_FORMAT)
  return `${moment(startDate).format(DATE_FORMAT)} - ${moment(endDate).format(DATE_FORMAT)}`
}

const Job = ({ enterprise, position, startDate, endDate, children }) => (
  <Container>
    <Title>{position} ({enterprise})</Title>
    <Period>
      <Calendar /> {formatPeriod(startDate, endDate)}
    </Period>
    {children}
  </Container>
)

Job.propTypes = {}
Job.defaultProps = {}

export default Job
