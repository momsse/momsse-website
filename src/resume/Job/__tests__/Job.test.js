import React from 'react'
import renderer from 'react-test-renderer'
import Job from '../index'

describe('<Job />', () => {
  it('should render successfully', () => {
    const tree = renderer
      .create(
        <Job
          enterprise="Chauffeur privÃ©"
          position="Software engineer"
          startDate={new Date('Nov 01 2015')}
        >
          Hello
        </Job>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
