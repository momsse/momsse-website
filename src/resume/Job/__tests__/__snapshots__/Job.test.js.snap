// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Job /> should render successfully 1`] = `
.c0 {
  padding: 0.5rem;
  background-color: #F0F0F0;
  border: 1px solid #d4d4d4;
  box-shadow: 0 1px 6px rgba(0,0,0,0.175);
}

.c1 {
  font-size: 1.25rem;
  margin-bottom: 0.25rem;
}

.c2 {
  color: #7f807a;
  font-size: 0.75rem;
  margin-bottom: 0.5rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Software engineer
     (
    Chauffeur priv√©
    )
  </div>
  <div
    className="c2"
  >
    <svg
      attr={
        Object {
          "viewBox": "0 0 448 512",
        }
      }
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
    >
      <path
        d="M436 160H12c-6.6 0-12-5.4-12-12v-36c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48v36c0 6.6-5.4 12-12 12zM12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm116 204c0-6.6-5.4-12-12-12H76c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-40zm0-128c0-6.6-5.4-12-12-12H76c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-40zm128 128c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-40zm0-128c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-40zm128 128c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-40zm0-128c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-40z"
      />
    </svg>
     
    novembre 2015
  </div>
  Hello
</div>
`;
